### DISCUSSION TITLE
link from rdoc to soure code?

### DISCUSSION THREAD

**Discussion by benkoshy** [jeremyevans/rodauth#252] [General]
For example I am on this page:

https://rodauth.jeremyevans.net/rdoc/files/doc/confirm_password_rdoc.html

Was wondering if there was any easy/quick way to navigate from the documentation directly to the source code?

Context: I am trying to work out what is the path required to hit the `confirm_password_view`, and i want to hit that endpoint.

**Comment by jeremyevans** [comment]
`confirm_password_path` should work.  Every `*_route` method (which you can configure), has a corresponding `*_path` method that returns the full path to the route. 

You cannot navigate directly to the source code in a lot of cases, because many of the internal methods are defined via metaprogramming (including `confirm_password_view`: https://github.com/jeremyevans/rodauth/blob/master/lib/rodauth/features/confirm_password.rb#L9).

**Reply by benkoshy** [reply]
Chrs  for the link - I should have pointed out that I did manage to find the source code, so i apologize for that - i wrote it merely as an idea / convenience for readers navigating from the rdocs. my 2 cents worth: while it may not be feasible to get a direct link to the method in the source code, a link to the module / feature would prob be helpful (esp. to new users)?

**Reply by jeremyevans** [reply]
That would be a lot of work to setup (there are over 1000 configuration methods that would have to be linked), and would make refactoring more difficult (adding metaprogramming would require changing all related links).  I do agree it would be helpful to new users, but I don't think the benefits are worth the costs.
